### PR TITLE
[ADXT-483] Add Azure AKS provisioner

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

This PR adds a new provisioner to create a Kubernetes environment using `AKS` on Azure.

The Provisioner should not be used yet because we have no way to clean resources. The provisioner is working fine with the example added in `test/new-e2e/examples`. But some more setup is required to have it working on the CI.
It can be addressed in a follow-up PR that would do the general setup for provisioner on datadog-agent

### Motivation

Run e2e tests on AKS 🎉

### CODE REVIEW

File: test/new-e2e/pkg/runner/ci_profile.go [hunk 1]
```diff
@@ -15,9 +15,13 @@ import (
 
 const (
 	defaultCISecretPrefix = "ci.datadog-agent."
-	defaultCIEnvironments = "aws/agent-qa"
 )
 
+var defaultCIEnvironments = map[string]string{
+	"aws": "agent-sandbox",
+	"az":  "agent-sandbox",
+}
```
Comment (pducolin) [inline test/new-e2e/pkg/runner/ci_profile.go:23]: **❓ question**
Should `defaultCIEnvironemnts` be `agent-qa` instead?
```suggestion
var defaultCIEnvironments = map[string]string{
	"aws": "agent-qa",
	"az":  "agent-qa",
}
```

File: test/new-e2e/pkg/runner/profile.go [hunk 2]
```diff
@@ -98,6 +99,24 @@ func newProfile(projectName string, environments []string, store parameters.Stor
 	return p
 }
 
+func defaultEnvironments(environments string, defaultsEnvironments map[string]string) string {
```
Comment (pducolin) [inline test/new-e2e/pkg/runner/profile.go:102]: **💬 suggestion**
Having `default` in both arguments and function can be confusing, I suggest a renaming and adding a comment to explain what this function does
```suggestion
// mergeEnvironments returns a string with a space separated list of available environments. It merges environments with a `defaultEnvironments` map
func mergeEnvironments(environments string, defaultsEnvironments map[string]string) string {
```

File: test/new-e2e/pkg/runner/profile.go [hunk 3]
```diff
@@ -98,6 +99,24 @@ func newProfile(projectName string, environments []string, store parameters.Stor
...
+		if len(parts) == 2 {
+			environmentsMap[parts[0]] = parts[1]
+		}
+	}
+	maps.Copy(defaultedEnvironmentsMap, environmentsMap)
+	defaultEnvironments := ""
+	for k, v := range defaultedEnvironmentsMap {
+		defaultEnvironments += fmt.Sprintf("%s/%s ", k, v)
+	}
+	return strings.TrimSpace(defaultEnvironments)
```
Comment (pducolin) [inline test/new-e2e/pkg/runner/profile.go:117]: I feel like we can skip one map. Also opinionated change to first copy default values and then overriding keys with elements from `environments`
**💬 suggestion**
```suggestion
	environmentsList := strings.Split(environments, " ")
	// set merged map capacity to worst case scenario of no overlapping key
	mergedEnvironmentsMap := make(map[string]string, len(defaultEnvironments) + len(environmentsList))
	maps.Copy(mergedEnvironmentsMap, defaultsEnvironments)
	for _, env := range environmentsList {
		parts := strings.Split(env, "/")
		if len(parts) == 2 {
			mergedEnvironmentsMap[parts[0]] = parts[1]
		}
	}
	maps.Copy(defaultedEnvironmentsMap, environmentsMap)
	mergedEnvironmentsList := make([]string, 0, len(mergedEnvironmentsMap))
	for k, v := range mergedEnvironmentsMap {
		mergedEnvironmentsList = mergedEnvironmentsList.append(mergedEnvironmentsList, fmt.Sprintf("%s/%s ", k, v))
	}
	return strings.Join(mergedEnvironmentsList, " ")
```

**Review by pducolin — APPROVED** [review]
Nice!

File: test/new-e2e/pkg/runner/profile.go (hunk 3 above)
Comment (pducolin) [reply test/new-e2e/pkg/runner/profile.go:118]: I should have done a single suggestion 🤦‍♀️

Comment (KevinFairise2) [reply test/new-e2e/pkg/runner/profile.go]: That's fine no worries! Applied them and I will fix the remaining things locally now!

**Pull request merged**